packages/tsconfig.json

```diff
@@ -0,0 +1,24 @@
+{
+	// Don't actually compile anything
+	"files": [],
+
+	// Reference all TS packages
+	"references": [
```

Comment:
It would be so cool to generate this list dynamically... but I don't know how :(

Reply:
If we're using the source directly, do we need this at all?

Reply:
I think so. These references is what makes it build types for all packages when we do `tsc --build packages/tsconfig.json`

Reply:
But do we need to run that? The client build uses the sources it needs. The packages know how to build themselves for publish. I don't understand where this fits… maybe it's important to typecheck all the packages simply on CI?

Reply:
My VSCode complained if `dist/types/*` don't exists, but now I can't replicate it anymore 🤷 
Deleted.

Reply:
If none of the packages' `dist` folders are not built, I can see that my IDE (Atom with TypeScript plugins) is confused:

(file _client/components/calypso-i18n-provider/index.tsx_)
<img width="776" alt="Screenshot 2020-11-18 at 09 04 27" src="https://user-images.githubusercontent.com/664258/99502677-b5a3e500-297d-11eb-99bb-49c9201fe1d9.png">

The `main`, `module`, `types` fields of the `react-i18n` package point to the `dist` directory which doesn't exist. And the TypeScript Language Server that backs the IDE features doesn't recognize the `calypso:src` field. So it can't find any of the `react-i18n` package's contents and can't typecheck the code that uses it.

This was already a concern in #43679 where I first tried this approach. Anything that uses the native `tsc` compiler will have trouble resolving the `calypso:src`-enabled packages.

At the same time, the `yarn start` task executed flawlessly and a working Calypso build is humming happily in the background. It's just the IDE that's broken.

Reply:
Yes! That's the same error I got. Maybe I can't reproduce it anymore because something related to caches?

Anyway, I think I should bring back the list of modules and the references.

Reply:
> Maybe I can't reproduce it anymore because something related to caches?

I think IDEs like VSCode or Atom run the [`tsserver`](https://github.com/Microsoft/TypeScript/wiki/Standalone-Server-%28tsserver%29) in background and it's the language server that really loads the source files, does the module resolution and typechecking. So maybe your server had already loaded the module's `dist` files and keeps them in memory, even after the `dist` folder on disk was cleaned?

Reply:
Thanks both, now I understand better. In summary, this is _redundant for the build_ but _helps editor integrations work better_. Happy to keep it 👍